rolando solution/assignment2CST438.py

```diff
@@ -0,0 +1,169 @@
+# all users and posts
+users = []
+posts = []
+
+# class hloding user info
+class User:
+  def __init__(self, username, displayName, state):
+    self.username = username
+    self.displayName = displayName
+    self.state = state
+    self.friends = []
+
+  def addFriends(self, friends):
+    self.friends.extend(friends)
+
+  def __str__(self):
+    return f'username: {self.username}\ndisplayName: {self.displayName}\nfriends: {self.friends}'
+  
+#class holding post info
+class Post:
+  def __init__(self, postId, username, visibility):
+    self.postId = postId
+    self.username = username
+    self.visibility = visibility
+  
+  def __str__(self):
+    return f'post id: {self.postId}\nusername: {self.username}\nvisibility: {self.visibility}'
+
+def processUsers(file):
+  usersInfo = file.readlines()
+  for userInfo in usersInfo:
+    # strip file line and split user info by ;
+    userInfo.strip()
+    userSplit = userInfo.split(';')
+
+    # add user basic info
+    user = User(userSplit[0], userSplit[1], userSplit[2])
+    users.append(user)
+
+    # adds friends to user
+    friendSection = userSplit[3].strip('[]\n')
+    friends = friendSection.split(',')
+    user.addFriends(friends)
+
+  return users
+
+def processPosts(file):
+  postsInfo = file.readlines()
+  for postInfo in postsInfo:
+    # strip line and add to post object
+    postInfo.strip()
+    postSplit = postInfo.split(';')
+    post = Post(postSplit[0], postSplit[1], postSplit[2].strip('\n'))
+    posts.append(post)
+
+  return posts
+
+# loading data from path
+def loadInputData():
```

Comment:
Separating and using recursion in your functions makes your code very readable and efficient.